### PR TITLE
Update Adam documentation

### LINKED ISSUE #41477: Adam implementation different from paper
In https://pytorch.org/docs/stable/_modules/torch/optim/adam.html#Adam, towards the end, we have

```
                    denom = (exp_avg_sq.sqrt() / math.sqrt(bias_correction2)).add_(group['eps'])
```

However, the [paper](https://arxiv.org/pdf/1412.6980.pdf) uses epsilon by adding it to `exp_avg_sq.sqrt()`; not to a whole fraction.
This is always the case, in both the Algorithm 1 formulation, as well as the one before Section 2.1.

What pytorch does is that it uses `epsilon * sqrt(1 - beta_2^t)` instead of just `epsilon`, for the paper's algoritm formulation.

cc @jlin27 @vincentqb

### PR DESCRIPTION
**Pull request by farhadrgh**
This PR fixes #41477

Adam implementation is doing L2 regularization and not decoupled weight decay. However, the change mentioned in #41477 was motivated by Line 12 of algorithm 2 in [Decoupled Weight Decay Regularization](https://arxiv.org/pdf/1711.05101.pdf) paper.

Please let me know if you have other suggestions about how to deliver this info in the docs.
cc @ezyang

### CODE REVIEW

**Review by vincentqb — APPROVED** [review]
LGTM, thanks!
